modules/task/backend.go

```diff
@@ -118,7 +93,28 @@ func (b *inMemBackend) Name() string {
 // Start implements the Module interface
 func (b *inMemBackend) Start(ready chan<- struct{}) <-chan error {
 	b.isRunning = true
-	return make(chan error)
+	errorCh := make(chan error, 2)
+	go func() {
+		for {
+			select {
+			case msg, ok := <-b.bufQueue:
+				if ok {
```

Comment:
No, it should just stop listening if the channel is closed. Added an else return here.